packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -642,16 +642,21 @@ contract Attestations is IAttestations, Ownable, Initializable, UsingRegistry, R
     internal
   {
     IRandom random = IRandom(registry.getAddressForOrDie(RANDOM_REGISTRY_ID));
+    IValidators validators = IValidators(
+      registry.getAddressForOrDie(VALIDATORS_REGISTRY_ID)
+    );
 
     bytes32 seed = random.random();
-    address[] memory validators = getValidators();
+    uint256 numberValidators = validators.numberValidatorsInCurrentSet();
 
     uint256 currentIndex = 0;
     address validator;
 
     while (currentIndex < n) {
       seed = keccak256(abi.encodePacked(seed));
-      validator = validators[uint256(seed) % validators.length];
+      validator = validators.validatorAddressFromCurrentSet(
```

Comment:
Do we want to store the `validator`, `account`, or `attestations` address here?

Personally, would vote for `account`, which would allow the validator to swap out their `attestations` key at any time without invalidating previous attestation requests.

@nambrot wdyt?

Reply:
Agreed, probably the account!

Reply:
Should be resolved in https://github.com/celo-org/celo-monorepo/pull/1313